hi-ml-multimodal/src/health_multimodal/image/model/model.py

```diff
@@ -74,7 +75,23 @@ class ImageModelOutput():
     projected_patch_embeddings: torch.Tensor
 
 
-class ImageModel(nn.Module):
+class MinimalImageModel(nn.Module):
+    """ An abstract class defining methods required by the ImageInferenceEngine """
+
+    def __init__(self, **kwargs: Any):
+        super().__init__()
+        assert hasattr(self, "training")
+
+    @abstractmethod
+    def forward(self, x: torch.Tensor) -> ImageModelOutput:
+        pass
```

Comment:
```suggestion
        raise NotImplementedError
```

(same for other methods)